gbpservice/neutron/nsf/configurator/drivers/vpn/vyos/vyos_vpn_driver.py

```diff
@@ -0,0 +1,642 @@
+                   "%r" % (url, str(err).capitalize()))
+            LOG.error(msg)
+            raise Exception(msg)
+        if not result['status']:
+            msg = ("Error adding persistent rule. URL: %r" % url)
+            LOG.error(msg)
+            raise Exception(msg)
+
+        msg = ("Persistent rule successfully added for SERVICE ID: %r"
+               " of tenant: %r" % (rule_info['service_id'],
+                                   rule_info['tenant_id']))
+        LOG.info(msg)
+
+    def clear_interfaces(self, context, kwargs):
+
+        rule_info = kwargs['rule_info']
+
+        active_rule_info = dict(
+            provider_mac=rule_info['provider_mac'],
+            stitching_mac=rule_info['stitching_mac'])
+
+        active_fip = rule_info['fip']
+
+        msg = ("Initiating DELETE persistent rule for SERVICE ID: %r of "
+               "tenant: %r " %
+               (rule_info['service_id'], rule_info['tenant_id']))
+        LOG.info(msg)
+        url = const.request_url % (active_fip,
+                                   const.CONFIGURATION_SERVER_PORT,
+                                   'delete_rule')
+
+        try:
+            data = json.dumps(active_rule_info)
+            resp = requests.delete(url, data=data, timeout=self.timeout)
+        except requests.exceptions.ConnectionError, err:
+            msg = ("Failed to establish connection to service at: %r "
+                   "of SERVICE ID: %r of tenant: %r . ERROR: %r" % (
+                    active_fip, rule_info['service_id'],
+                    rule_info['tenant_id'], str(err).capitalize()))
+            LOG.error(msg)
+            raise Exception(err)
+        except requests.exceptions.RequestException, err:
+            msg = ("Unexpected ERROR happened  while deleting "
+                   "persistent rule of service at: %r "
+                   "of SERVICE ID: %r of tenant: %r . ERROR: %r" % (
+                    active_fip, rule_info['service_id'],
+                    rule_info['tenant_id'], str(err).capitalize()))
+            LOG.error(msg)
+            raise Exception(err)
+
+        try:
+            result = resp.json()
+        except ValueError as err:
+            msg = ("Unable to parse response, invalid JSON. URL: "
+                   "%r" % (url, str(err).capitalize()))
+            LOG.error(msg)
+            raise Exception(msg)
+        if not result['status'] or resp.status_code not in [200, 201, 202]:
+            msg = ("Error deleting persistent rule. URL: %r" % url)
+            LOG.error(msg)
+            raise Exception(msg)
+        msg = ("Persistent rule successfully deleted for SERVICE ID: %r"
+               " of tenant: %r " % (rule_info['service_id'],
+                                    rule_info['tenant_id']))
+
+
+class VpnaasIpsecDriver(VpnGenericConfigDriver, BaseDriver):
+    """
+    Driver class for implementing VPN IPSEC configuration
+    requests from VPNaas Plugin.
+    """
+    service_type='vpn'
```

Comment:
Take service type from constants. If this is single constant for entire VPN, keep it in this file itself,